### PR TITLE
[To Feature ] DESENG-605 - Adding Unit test

### PR DESCRIPTION
**Pull request by ratheesh-aot**
Issue #: https://apps.itsm.gov.bc.ca/jira/browse/DESENG-605

*Description of changes:*
Added unit test for Tenant listing and Tenant Detail page

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-web/jest.config.ts [hunk 1]
```diff
@@ -188,6 +188,8 @@ const config: Config.InitialOptions = {
 
     // Whether to use watchman for file crawling
     // watchman: true,
+
+    testTimeout: 10000,
```
Comment (NatSquared) [inline met-web/jest.config.ts:192]: Were these tests timing out when running locally?

File: met-web/tests/unit/components/tenantManagement/TenantDetail.test.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        render(
+            <MemoryRouter initialEntries={['/tenantadmin/1/detail']}>
+                <Routes>
+                    <Route path="/tenantadmin/:tenantId/detail" element={<TenantDetail />} />
+                </Routes>
+            </MemoryRouter>,
+        );
+
+        const loadingTexts = screen.getAllByText('Loading...');
+        expect(loadingTexts.length).toBeGreaterThan(0); // Adjust based on expected occurrences
```
Comment (NatSquared) [inline met-web/tests/unit/components/tenantManagement/TenantDetail.test.tsx:125]: Was this left in intentionally?

File: met-web/tests/unit/components/tenantManagement/TenantDetail.test.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+        await waitFor(() => {
+            const tenantNames = screen.getAllByText('Tenant One');
+            expect(tenantNames).toHaveLength(2);
+            expect(screen.getByText('Title One')).toBeVisible();
+            expect(screen.getByText('Description One')).toBeVisible();
+            expect(screen.getByText('Contact One')).toBeVisible();
+            expect(screen.getByText('contactone@example.com')).toBeVisible();
+            expect(screen.getByText('Photographer One')).toBeVisible();
+            expect(screen.getByText('Logo Description One')).toBeVisible();
```
Comment (NatSquared) [inline met-web/tests/unit/components/tenantManagement/TenantDetail.test.tsx:109]: I appreciate how thorough these tests are :)

**Review by NatSquared — APPROVED** [review]
Thanks for your work here! Looks good; approved

File: met-web/tests/unit/components/tenantManagement/TenantDetail.test.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
+import React, { ReactNode } from 'react';
+import { render, waitFor, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import * as reactRedux from 'react-redux';
+import * as reactRouter from 'react-router';
+import * as tenantService from 'services/tenantService';
+import TenantDetail from '../../../../src/components/tenantManagement/Detail';
+import { MemoryRouter, Route, Routes, useParams } from 'react-router-dom';
```
Comment (NatSquared) [inline met-web/tests/unit/components/tenantManagement/TenantDetail.test.tsx:8]: Just noticed this unused import :p
```suggestion
import { MemoryRouter, Route, Routes} from 'react-router-dom';
```

File: met-web/tests/unit/components/tenantManagement/TenantListing.test.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
+import React, { ReactNode } from 'react';
+import { render, waitFor, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { setupEnv } from '../setEnvVars';
+import * as reactRedux from 'react-redux';
+import * as reactRouter from 'react-router';
+import * as tenantService from 'services/tenantService';
+import { draftEngagement } from '../factory';
```
Comment (NatSquared) [inline met-web/tests/unit/components/tenantManagement/TenantListing.test.tsx:8]: this is unused too 🙃 
```suggestion
```

File: met-web/jest.config.ts (hunk 1 above)
Comment (ratheesh-aot) [reply met-web/jest.config.ts:192]: sometimes old tests also timing out locally

File: met-web/tests/unit/components/tenantManagement/TenantDetail.test.tsx (hunk 2 above)
Comment (ratheesh-aot) [reply met-web/tests/unit/components/tenantManagement/TenantDetail.test.tsx]: Removed

File: met-web/tests/unit/components/tenantManagement/TenantDetail.test.tsx (hunk 4 above)
Comment (ratheesh-aot) [reply met-web/tests/unit/components/tenantManagement/TenantDetail.test.tsx]: Removed

File: met-web/tests/unit/components/tenantManagement/TenantListing.test.tsx (hunk 5 above)
Comment (ratheesh-aot) [reply met-web/tests/unit/components/tenantManagement/TenantListing.test.tsx]: Removed

**Pull request merged**